radicle-cli/src/commands/remote/add.rs

```diff
@@ -0,0 +1,56 @@
+use radicle::node::tracking::store::Config;
+use radicle::{git::Url, node::TRACKING_DB_FILE, prelude::Id, Profile};
+use radicle_crypto::PublicKey;
+
+use crate::{git, terminal as term};
+
+/// core command to run the `rad remote add ...` subcommand.
+pub fn run(
+    repository: &git::Repository,
+    profile: &Profile,
+    pubkey: &PublicKey,
+    name: &Option<String>,
+    id: Id,
+) -> anyhow::Result<()> {
+    let name = if let Some(name) = name {
+        name.to_owned()
+    } else if let Some(name) = lookup_alias(profile, pubkey)? {
+        name
+    } else {
+        anyhow::bail!("an alias need to be specified");
+    };
+
+    if git::rad_has_remote(repository, &name)? {
+        anyhow::bail!("remote with did `{pubkey}` already present");
+    }
+    let url = Url::from(id).with_namespace(*pubkey);
+    let (name, url) = add_new_remote(repository, &name, &url)?;
+
+    term::success!("Remote {name} added with {url}");
+    Ok(())
+}
+
+/// from a node pubkey try to get the alias of the node
+fn lookup_alias(profile: &Profile, pubkey: &PublicKey) -> anyhow::Result<Option<String>> {
+    let path = profile.home.node().join(TRACKING_DB_FILE);
+    let storage = Config::reader(path)?;
+    let Some(node) = storage.node_policy(pubkey)? else {
+        return Ok(None);
+    };
+    Ok(node.alias)
+}
+
+fn add_new_remote(
+    repository: &git::Repository,
+    alias: &str,
+    url: &Url,
+) -> anyhow::Result<(String, String)> {
+    let remote = repository.remote(alias, &url.to_string())?;
+    Ok((
+        remote.name().unwrap_or_default().to_owned(),
+        remote
+            .url()
+            .ok_or(anyhow::anyhow!("remote url not present"))?
+            .to_owned(),
+    ))
+}
```

Comment:
I don't understand what the purpose of this function is -- just call `repo.remote(..)`?

Reply:
Yeah there is no reason, I think during the iteration this method change :) so I think this has no sense at all now